sdk/src/main/scala/com/horizen/block/SidechainsHashMap.scala

```diff
@@ -132,26 +152,39 @@ class SidechainHashMap
   }
 
   def getMerklePath(sidechaiId: ByteArrayWrapper): Option[MerklePath] = {
-    sidechainHashMap.get(sidechaiId) match {
+    sidechainsHashMap.get(sidechaiId) match {
       case Some(shl) =>
         val mkt = getMerkleTree
-        val mkr = new ByteArrayWrapper(getMerkleRoot(sidechaiId))
-        val leafIndex = mkt.leaves().asScala.map(l => new ByteArrayWrapper(l)).lastIndexOf(mkr)
+        val mkr = new ByteArrayWrapper(getSidechainHash(sidechaiId))
+        val leafIndex = mkt.leaves().asScala.map(l => new ByteArrayWrapper(l)).indexOf(mkr)
         Some(mkt.getMerklePathForLeaf(leafIndex))
       case None => None
     }
   }
 
-  def getNeighborsMerklePaths(sidechainId: ByteArrayWrapper): (Option[(Array[Byte], MerklePath)], Option[(Array[Byte], MerklePath)]) = {
-    val scl: Seq[ByteArrayWrapper] = (sidechainHashMap.map(_._1).toSeq :+ sidechainId).sortWith(_ < _)
-    val leftNeighborOption = Try(scl(scl.indexOf(sidechainId) - 1))
+  def getNeighborProofs(sidechainId: ByteArrayWrapper): (Option[NeighbourProof], Option[NeighbourProof]) = {
```

Comment:
Both Neighbor and Neighbo**u**r word variants used now. Use only one everywhere.